src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -106,69 +106,76 @@ uct_cuda_ipc_md_query(uct_md_h md, uct_md_attr_v2_t *md_attr)
 }
 
 static ucs_status_t
-uct_cuda_ipc_mem_reg_push_ctx(CUdeviceptr address, CUdevice *cuda_device_p,
-                              int *is_ctx_pushed)
+uct_cuda_ipc_memh_init(void *address, uct_cuda_ipc_memh_t *memh)
 {
 #define UCT_CUDA_IPC_NUM_ATTRS 2
-    CUdevice cuda_device = CU_DEVICE_INVALID;
     CUpointer_attribute attr_type[UCT_CUDA_IPC_NUM_ATTRS];
     void *attr_data[UCT_CUDA_IPC_NUM_ATTRS];
-    CUcontext cuda_ctx, cuda_curr_ctx;
     int cuda_device_ordinal;
     ucs_status_t status;
 
-    *is_ctx_pushed = 0;
-
     attr_type[0] = CU_POINTER_ATTRIBUTE_CONTEXT;
-    attr_data[0] = &cuda_ctx;
+    attr_data[0] = &memh->cuda_ctx;
```

Comment:
I'd prefer to change the output parameters only in successful flow. And use local variables to temporarily store obtained data.
The same comment is valid for `mkey_pack` call chain.